infrastructure/ansible/roles/dataset_loader/tasks/dataset_loader.yml

```diff
@@ -368,22 +368,21 @@
 
 - name: Assemble assignments of region to division
   set_fact:
-    region_assignments: "{
-                        {%- for d in dl_ds_merged_divisions -%}
-                        {%- set outer_loop = loop -%}
-                        {%- for r in d.regions -%}
-                        '{{ r.region_name }}': '{{ d.division_name }}'
-                        {%- if not loop.last %},{% endif -%}
-                        {%- endfor %}
-                        {%- if not outer_loop.last %},{% endif -%}
-                        {%- endfor %}
-                        }"
+    region_assignments: "{{ (region_assignments | default({})) | combine(regions) }}"
+  with_items: "{{ get_all_divisions.json.response }}"
+  vars:
+    region_query: "[?division_name == `{{ item.name }}`].regions[].{region_name: region_name, division_id: `{{ item.id }}`}"
+    regions: "{{ dl_ds_merged_divisions | to_json | from_json | json_query(region_query) | items2dict(key_name='region_name',value_name='division_id') }}"
+
 - name: Create Regions
   uri:
-    url: "{{ dl_to_url }}/api/{{ dl_to_api_version }}/divisions/{{ item.value }}/regions"
+    url: "{{ dl_to_url }}/api/{{ dl_to_api_version }}/regions"
     method: POST
-    body:
-      name: "{{ item.key }}"
+    body: >
+      {
+      "name": "{{ item.key }}",
+      "division": {{ item.value | int}}
+      }
```

Comment:
This kind of pattern isn't recommended, but it is necessary either inline like this or a J2 template file because the to_json/to_nice_json filters are weakly typed by default.